connector/apmconnector/metric_to_metric_connector.go

```diff
@@ -0,0 +1,257 @@
+// Copyright New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package apmconnector // import "github.com/newrelic/opentelemetry-collector-components/connector/apmconnector"
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+)
+
+type OpenTelemetryMetricToApmMetricConnector struct {
+	config *Config
+	logger *zap.Logger
+
+	metricsConsumer consumer.Metrics
+}
+
+func (c *OpenTelemetryMetricToApmMetricConnector) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: false}
+}
+
+func (c *OpenTelemetryMetricToApmMetricConnector) ConsumeMetrics(ctx context.Context, td pmetric.Metrics) error {
+	metrics := ConvertMetrics(c.logger, c.config, td)
+	return c.metricsConsumer.ConsumeMetrics(ctx, metrics)
+}
+
+func (c *OpenTelemetryMetricToApmMetricConnector) Start(_ context.Context, _ component.Host) error {
+	c.logger.Info("Starting the APM Metric Connector")
+	if c.config.ApdexT == 0 {
+		c.config.ApdexT = defaultApdexT
+	}
+	return nil
+}
+
+func (c *OpenTelemetryMetricToApmMetricConnector) Shutdown(context.Context) error {
+	c.logger.Info("Stopping the APM Metric Connector")
+	return nil
+}
+
+func ConvertMetrics(logger *zap.Logger, config *Config, md pmetric.Metrics) pmetric.Metrics {
+	apdex := NewApdex(config.ApdexT)
+	newMetrics := pmetric.NewMetrics()
+	attributesFilter := NewAttributeFilter()
+	metricMap := NewMetrics()
+
+	for i := 0; i < md.ResourceMetrics().Len(); i++ {
+		rm := md.ResourceMetrics().At(i)
+
+		rmNew := pmetric.ResourceMetrics{}
+		metrics := &ResourceMetrics{}
+
+		for j := 0; j < rm.ScopeMetrics().Len(); j++ {
+			sm := rm.ScopeMetrics().At(j)
+
+			smNew := pmetric.ScopeMetrics{}
+
+			for k := 0; k < sm.Metrics().Len(); k++ {
+				m := sm.Metrics().At(k)
+
+				if m.Name() == "http.server.duration" || m.Name() == "http.server.request.duration" {
```

Comment:
These two metrics are the same, but `http.server.request.duration` is the newer name, correct? Might be worth adding a comment.